### PR TITLE
DCOS-2262 Send hostname segment.io track events

### PR DESCRIPTION
**Pull request by jsancio**
Verified that the value gets sent by doing manual tests.

### CODE REVIEW

File: cli/dcoscli/analytics.py [hunk 1]
```diff
@@ -261,10 +262,18 @@ def _base_properties(conf=None):
         cmd = 'dcos'
         full_cmd = 'dcos'
 
+    try:
+        dcos_hostname = six.moves.urllib.parse.urlparse(
+            conf.get('core.dcos_url')).hostname
+    except:
```
Comment (mgummelt) [inline cli/dcoscli/analytics.py]: except DCOSException

File: cli/dcoscli/analytics.py [hunk 2]
```diff
@@ -261,10 +262,18 @@ def _base_properties(conf=None):
...
+        dcos_hostname = None
+
     return {
         'cmd': cmd,
         'full_cmd': full_cmd,
         'dcoscli.version': dcoscli.version,
         'python_version': str(sys.version_info),
-        'config': json.dumps(list(conf.property_items()))
+        'config': json.dumps(list(conf.property_items())),
+        'DCOS_HOSTNAME': dcos_hostname,
```
Comment (mgummelt) [inline cli/dcoscli/analytics.py]: plz stay consistent with lower case

**Comment by mgummelt** [comment]
I think this will make test_analytics fail

File: cli/dcoscli/analytics.py (hunk 2 above)
Comment (jsancio) [inline cli/dcoscli/analytics.py]: I didn't pick the key name. The analytic team want it this way so they can do a `group by`.

File: cli/dcoscli/analytics.py (hunk 1 above)
Comment (jsancio) [inline cli/dcoscli/analytics.py]: Why DCOSException? `urlparse` can fail.

Comment (mgummelt) [inline cli/dcoscli/analytics.py]: nvm

**Comment by jsancio** [comment]
@mgummelt Are we happy? I want get this in before Friday.

**Comment by mgummelt** [comment]
LGTM

**Comment by jsancio** [comment]
Great! Thanks.

**Pull request merged**